### PR TITLE
DLUHC-204 add map component

### PR DESCRIPTION
**Pull request by samornsby**
* Adds a basic map component and context container that we can build upon.
* Renders an example map on the index page
image

### CODE REVIEW

File: dluhc-component-library/src/components/maps/baseMap.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    useGeographic();
+    map.setLayers([new TileLayer({ source: new OSM() })]);
+    map.setView(
+      new View({
+        center: [lng, lat],
+        zoom,
+      }),
+    );
+    map.setTarget(mapId);
+  }, [lng, lat, mapId, zoom]);
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/maps/baseMap.tsx:40]: map should also be in the array. Probably need to get eslint set up properly to catch this for us

File: dluhc-component-library/src/contexts/mapContext.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Map as OLMap } from "ol";
+import { ComponentChildren, createContext } from "preact";
+import { useContext } from "preact/hooks";
+
+const MapContext = createContext<OLMap | null>(null);
+
+function useMap() {
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/contexts/mapContext.tsx:7]: `Non-Blocker` Should probably be an arrow function

File: dluhc-component-library/src/components/maps/baseMap.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+interface BaseMapProps {
+  mapId: string;
+  lat?: number;
+  lng?: number;
+  zoom?: number;
+  className?: string;
+  style?: CSSProperties;
+}
+
+function BaseMap({
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/components/maps/baseMap.tsx:19]: `Non-Blocker` function is fine but we should probably be consistent across the repo (this PR is consistent) and use arrow functions if we don't need to make use of `this`

File: dluhc-component-library/src/main.tsx [hunk 4]
```diff
@@ -18,7 +19,23 @@ const renderSiteSelectionForm: renderFunction = (
   render(<SiteSelectionForm />, element);
 };
 
+const renderMap: renderFunction = (_options: {}, element: HTMLElement) => {
+  render(
+    <MapContainer>
+      <BaseMap
+        mapId="site-selection-map"
+        lat={54.97}
```
Comment (amjones-scottlogic) [inline dluhc-component-library/src/main.tsx:27]: `Non-Blocker` Doesnt need to be part of this PR but is it worth pulling this out into the render function options?

File: dluhc-component-library/src/components/maps/baseMap.tsx (hunk 1 above)
Comment (samornsby) [reply dluhc-component-library/src/components/maps/baseMap.tsx:40]: Agreed, the lint rules would be good to have

File: dluhc-component-library/src/main.tsx (hunk 4 above)
Comment (samornsby) [reply dluhc-component-library/src/main.tsx:39]: Will do this when we have a more specific map implementation. This render function will probably be removed in the near future.

**Review by amjones-scottlogic — APPROVED** [review]

**Pull request merged**